cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/AnomalyDetector.java

```diff
@@ -284,8 +288,7 @@ private boolean isReadyToFix(Anomaly anomaly) {
         LOG.info("Skipping {} because load monitor is in {} state.", skipMsg, loadMonitorTaskRunnerState);
         _anomalyDetectorState.onAnomalyHandle(anomaly, AnomalyState.Status.LOAD_MONITOR_NOT_READY);
       } else {
-        boolean meetCompletenessRequirements = _kafkaCruiseControl.meetCompletenessRequirements(Collections.emptyList());
-        if (meetCompletenessRequirements) {
+        if (_kafkaCruiseControl.meetCompletenessRequirementsWithDefaultGoals()) {
```

Comment:
Since we add `_selfHealingGoals` field to `BrokerFailures` and `GoalViolations`, I feel we should not unconditionally check completeness against default goal list here.

 Currently this is ok since we always set `_selfHealingGoals` to empty list and default goal list will be picked up. I think the reason we add `_selfHealingGoals` is because we may want to tune the goals use for self-healing in the future and this implicit assumption will not hold?